Extend service invocation to non-Dapr endpoints docs (#3384)
dapr/docs: daprdocs/content/en/developing-applications/building-blocks/service-invocation/service-invocation-overview.md

Using service invocation, your application can reliably and securely communicate with other applications using the standard [gRPC](https://grpc.io) or [HTTP](https://www.w3.org/Protocols/) protocols.

> In many microservice-based applications multiple services need the ability to communicate with one another, as well as with non-Daprized endpoints. This inter-service, and external communication requires that application developers handle problems like:

Reviewer: Scrap this change. Just add a new section with a diagram with the intro applies to all service invocation scenarios

Author: I reverted back to the original sentence, but removed the `inter-service`  word specification, because I don't want it to be thought of as only internal Dapr communication. Also, please lmk what you think on the additional diagram I added specific to non-Dapr endpoint communication.

- Each application communicates with its own instance of Dapr. 
- The Dapr instances discover and communicate with each other.

> The diagram below is an overview of how Dapr's service invocation works between two Daprized applications.

Reviewer 2: ```suggestion
The diagram below is an overview of how Dapr's service invocation works between two Dapr-ized applications.
```

6. Dapr forwards the response to Service A's Dapr sidecar.
7. Service A receives the response.

> The diagram below is an overview of how Dapr's service invocation works when invoking non-Daprized endpoints.

Reviewer: Move all of this detailed steps below along with the diagram into the "howto-invoke-non-daprized-endpoints.md" topic. Too much detail for an overview

Instead here
1) Put a sentence "You can also call non-Dapr HTTP endpoints using the service invocation API. For example you may only use Dapr in part of an overall application, you may not have access to the code to migrate an existing application to use Dapr, or you simply need to call an external HTTP service. Read [How-To: Invoke Non-Daprized Endpoints using HTTP]({{< ref XXXX >}}) for more information"
1) Then keep a version of the diagram without the numbers on to illustrate this.

Author: I moved everything to the other file like you recommended, but my thinking is do we want to include inconsistency in diagrams by me duplicating my diagram for non-Dapr endpoint invocation without the numbers for the overview page? I'm not the biggest fan of inconsistencies like that. I like the change to just a paragraph here, but if they click to then look for more information on non-Dapr endpoint invocation, then would it be sufficient to just have the diagram there, instead of duplicating without numbers on the service invocation overview page?

<img src="/images/service-invocation-overview-non-daprized-endpoint.png" width=800 alt="Diagram showing the steps of service invocation to non-Daprized endpoints">

1. Service A makes an HTTP call targeting Service B, a non-Daprized endpoint. The call goes to the local Dapr sidecar.

> 2. Dapr discovers Service B's location using the [name resolution component]({{< ref supported-name-resolution >}}) which is running on the given [hosting platform]({{< ref "hosting" >}}).

Reviewer: The Naming Component is not used here, that is for appId. You should update this to talk about resolving with either a name HTTPEndpoint resource or FQDM URL. I would also update this to have an picture that is a HTTP resource

Author: I updated... however, I mostly just removed the name resolution component reference. My thinking is Service B is simply a non-Dapr service, so it is still fair game to visualize it as a "Service" icon, but the description now denotes this to be discovered either using an HTTPEndpoint or FQDN URL.

3. Dapr forwards the message to Service B.
    - **Note**: Calls to non-Daprized endpoints use the HTTP protocol.
4. Service B runs its business logic code.
5. Service B sends a response to Service A's Dapr sidecar.
6. Service A receives the response.

## Features

> Service invocation provides several features to make it easy for you to call methods between applications, as well as to non-Daprized endpoints.

Reviewer: ```suggestion
Service invocation provides several features to make it easy for you to call methods between applications or to call external HTTP endpoints.
```

### HTTP and gRPC service invocation
- **HTTP**: If you're already using HTTP protocols in your application, using the Dapr HTTP header might be the easiest way to get started. You don't need to change your existing endpoint URLs; just add the `dapr-app-id` header and you're ready to go. For more information, see [Invoke Services using HTTP]({{< ref howto-invoke-discover-services.md >}}). 
- **gRPC**: Dapr allows users to keep their own proto services and work natively with gRPC. This means that you can use service invocation to call your existing gRPC apps without having to include any Dapr SDKs or include custom gRPC services. For more information, see the [how-to tutorial for Dapr and gRPC]({{< ref howto-invoke-services-grpc.md >}}).

> ### HTTP service invocation to non-Daprized endpoints

Reviewer: ```suggestion
### HTTP service invocation to external services or non-Dapr endpoints
```

> If you are in a brownfield scenario or simply have existing non-Daprized endpoints that you want to invoke, then you can do so using the service invocation API.

Reviewer: ```suggestion
There are times when you need to call an HTTP endpoint which is not using Dapr. For example you may choose to only use Dapr in part of you overall application, including brownfield development, you may not have access to the code to migrate an existing application to use Dapr, or you simply need to call an external HTTP service.  In these cases you can use the service invocation API.

```

> By defining a Dapr HTTPEndpoint, you declaratively define a way to interact with a non-Daprized endpoint. You then use the existing service invocation URL, while replacing the internal `appid` with the HTTPEndpoint name, and can use the existing service invocation API as is. This allows for you to also use the HTTPEndpoint name as the value if you use the `dapr-app-id` header. Alternatively, you can place a non-Daprized endpoint URL directly into the service invocation URL where you would place the `appid` or HTTPEndpoint name.

Reviewer: Move this into the How To Topic. This is where the details are needed. Also rewrite just to simply say how to use the HTTPEndpoint or FQDM URL and not try to say this is a replacement for appId. Blend this in with the order of precedence detail in the How To Topic.

```suggestion
By defining a HTTPEndpoint resource, you declaratively define a way to interact with a non-Daprized endpoint. 
```
